integration/tests/basic/indexer/test_events_by_type.py

```diff
@@ -0,0 +1,335 @@
+from collections import Counter
+
+import pytest
+
+import allure
+from integration.tests.basic.helpers.basic import NeonEventType
+from integration.tests.basic.helpers.rpc_checks import (
+    assert_event_by_type,
+    assert_event_field,
+    assert_events_order,
+    count_events,
+)
+from utils.accounts import EthAccounts
+from utils.models.result import NeonGetTransactionResult
+from utils.web3client import NeonChainWeb3Client
+
+
+@allure.feature("JSON-RPC validation")
+@allure.story("Verify events")
+@pytest.mark.usefixtures("accounts", "web3_client")
+class TestNeonEventType:
+    web3_client: NeonChainWeb3Client
+    accounts: EthAccounts
+
+    @pytest.mark.mainnet
+    @pytest.mark.neon_only
+    def test_event_log(self, json_rpc_client, event_caller_contract):
+        sender_account = self.accounts[0]
+        tx = self.web3_client.make_raw_tx(sender_account)
+        instruction_tx = event_caller_contract.functions.nonArgs().build_transaction(tx)
+        resp = self.web3_client.send_transaction(sender_account, instruction_tx)
+        response = json_rpc_client.send_rpc(method="neon_getTransactionReceipt", params=[resp["transactionHash"].hex()])
+        validated_response = NeonGetTransactionResult(**response)
+
+        events_counter = count_events(validated_response)
+        assert events_counter == Counter({"EnterCall": 1, "ExitStop": 1, "Return": 1, "Log": 1})
+        assert_event_field(validated_response, NeonEventType.Log, "address", validated_response.result.to)
+        assert_event_by_type(validated_response, NeonEventType.Log)
+        assert_event_by_type(validated_response, NeonEventType.EnterCall)
+        assert_event_by_type(validated_response, NeonEventType.ExitStop)
+        assert_event_by_type(validated_response, NeonEventType.Return)
+
+    @pytest.mark.mainnet
+    @pytest.mark.neon_only
+    def test_event_enter_call(self, json_rpc_client, event_caller_contract):
+        sender_account = self.accounts[0]
+        tx = self.web3_client.make_raw_tx(sender_account)
+        instruction_tx = event_caller_contract.functions.nonArgs().build_transaction(tx)
+        resp = self.web3_client.send_transaction(sender_account, instruction_tx)
+        response = json_rpc_client.send_rpc(method="neon_getTransactionReceipt", params=[resp["transactionHash"].hex()])
```

Comment:
let's move it to json_rpc_client to the new function like get_neon_trx_receipt

Reply:
Done ✅